text/0030-standardized-sender-sig.md

```diff
@@ -0,0 +1,162 @@
+- Feature Name: Standardized Sender Signature
+- Start Date: 2022-03-14
+- MCIP PR: [mobilecoinfoundation/mcips#0030](https://github.com/mobilecoinfoundation/mcips/pull/0030)
+- MobileCoin Epic: None
+
+# Summary
+[summary]: #summary
+
+Enact MCIP #18 (Janus Mitigation), which sets the ephemeral key in the encrypted fog hint
+to be the tx private key times the Ristretto base point. The original motivation was to enact
+Janus attack mitigation.
+
+We find a secondary use-case to motivate this: it becomes possible for the signer to create
+a Schnorr signature over any payload, which a verifier can validate against the keys appearing
```

Comment:
To make sure I'm following this: the public key that a sender would sign against is in the encrypted fog hint, so only the recipient's fog server can obtain this key and use it to verify messages signed with it?

Reply:
so, the way that cryptobox works is:

* I get your public key (a Ristretto curve point)
* I choose a random scalar (private fog key, `r_fog` in Koe's notation in Mechanics of Mobilecoin)
* I do key exchange against your public key and `r_fog`, and I put `r_fog * G` as the first 32 bytes of the e_fog_hint.
* I use the key resulting from key exchange to encrypt and mac my payload
* I append 2 versioning bytes.

So the public key that a sender would sign against is `r_fog`, and can be validated against `r_fog * G`, which is the first 32 bytes of fog hint.

Koe proposed that instead of having two independent random keys `r` and `r_fog`, we should just set `r_fog = r`, and this enables a Janus mitigation test, without really impacting fog since the `r_fog` number is a perfectly fine secret key anyways.

So this breaks some encapsulation that's separating the `e_fog_hint` field from the rest of the Tx math. But, it serves several useful purposes, and it's not that big a deal from the point of cryptobox / fog.